examples/cloud-run/tgi-deployment/README.md

```diff
@@ -0,0 +1,358 @@
+
+```bash
+export PROJECT_ID=your-project-id
+export LOCATION=your-location
+export CONTAINER_URI=us-docker.pkg.dev/deeplearning-platform-release/gcr.io/huggingface-text-generation-inference-cu121.2-2.ubuntu2204.py310
+export SERVICE_NAME=text-generation-inference
+```
+
+Then you need to login into your Google Cloud account and set the project ID you want to use to deploy Cloud Run.
+
+```bash
+gcloud auth login
+gcloud auth application-default login  # For local development
+gcloud config set project $PROJECT_ID
+```
+
+Once you are logged in, you need to enable the Cloud Run API, which is required for the Hugging Face DLC for TGI deployment on Cloud Run.
+
+```bash
+gcloud services enable run.googleapis.com
+```
+
+## Deploy TGI on Cloud Run
+
+Once you are all set, you can call the `gcloud beta run deploy` command (still on beta because GPU support is on preview, as mentioned above).
+
+The `gcloud beta run deploy` command needs you to specify the following parameters:
+
+- `--image`: The container image URI to deploy.
+- `--args`: The arguments to pass to the container entrypoint, being `text-generation-launcher` for the Hugging Face DLC for TGI. Read more about the supported arguments at [Text-generation-launcher arguments](https://huggingface.co/docs/text-generation-inference/en/basic_tutorials/launcher).
+  - `--model-id`: The model ID to use, in this case, [`hugging-quants/Meta-Llama-3.1-8B-Instruct-AWQ-INT4`](https://huggingface.co/hugging-quants/Meta-Llama-3.1-8B-Instruct-AWQ-INT4).
+  - `--quantize`: The quantization method to use, in this case, `awq`. If not specified, it will be retrieved from the `quantization_config->quant_method` in the `config.json` file.
+- `--port`: The port the container listens to.
+- `--cpu` and `--memory`: The number of CPUs and amount of memory to allocate to the container. Needs to be set to 8 and 32GiB, respectively; as that's a requirement for using the GPU.
+- `--no-cpu-throttling`: Disables CPU throttling, which is required for using the GPU.
+- `--gpu` and `--gpu-type`: The number of GPUs and the GPU type to use. Needs to be set to 1 and `nvidia-l4`, respectively; as at the time of writing this tutorial, those are the only available options as Cloud Run on GPUs is still under preview.
+- `--max-instances`: The maximum number of instances to run, set to 1 but default value is 7.
+- `--region`: The region to deploy the Cloud Run service.
+- `--no-allow-unauthenticated`: Disables unauthenticated access to the service, which is a good practice as adds an authentication layer managed by Google Cloud IAM.
+
+```bash
+gcloud beta run deploy $SERVICE_NAME \
+    --image=$CONTAINER_URI \
+    --args="--model-id=hugging-quants/Meta-Llama-3.1-8B-Instruct-AWQ-INT4,--quantize=awq" \
+    --port=8080 \
+    --cpu=8 \
+    --memory=32Gi \
+    --no-cpu-throttling \
+    --gpu=1 \
+    --gpu-type=nvidia-l4 \
+    --max-instances 1 \
```

Comment:
max instances = 1 can lead to downtime during infrastructure migrations. I would recommend to use 3.